imsim/lsst_image.py

```diff
@@ -57,8 +58,9 @@ def setup(self, config, base, image_num, obj_num, ignore, logger):
             self.vignetting = None
 
         self.apply_sky_gradient = params.get('apply_sky_gradient', False)
-
+        self.apply_fringing = params.get('apply_fringing',False)
         self.camera_name = params.get('camera')
+        self.boresight = params.get('boresight')
```

Comment:
boresight is required if apply_fringing is True.  You need to check that and raise an exception if not.

Reply:
And also test that the error is raised if the user messes this up.